File: Backend/user-service/spring_boot_backend_template/spring_boot_backend_template/src/main/java/com/backend/entity/SeatBooking.java
package com.backend.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "seat_bookings")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class SeatBooking {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    // Which journey (date + route)
    @ManyToOne
    @JoinColumn(name = "schedule_id", nullable = false)
    private BusSchedule busSchedule;

    // Which seat (1A, 1B, etc.)
    @ManyToOne
    @JoinColumn(name = "seat_id", nullable = false)
    private Seat seat;

    @Column(nullable = false)
    private boolean booked; // true = seat is booked
}

File: Backend/admin-service/admin/admin-service/Models/SeatBooking.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace admin_service.Models
{
    [Table("seat_bookings")]
    public class SeatBooking
    {
        public long Id { get; set; }

        [Column("booked")]
        public bool Booked { get; set; }

        // 👇 Explicit FK mapping (THIS FIXES THE ERROR)
        [Column("schedule_id")]
        public long ScheduleId { get; set; }

        [ForeignKey(nameof(ScheduleId))]
        public BusSchedule BusSchedule { get; set; }

        // 👇 Explicit FK mapping
        [Column("seat_id")]
        public long SeatId { get; set; }

        [ForeignKey(nameof(SeatId))]
        public Seat Seat { get; set; }
    }
}
